src/NuGet.Core/NuGet.Common/InProcLock.cs

```diff
@@ -0,0 +1,156 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace NuGet.Common
+{
+    internal class InProcLock : IDisposable
```

Comment:
This type, while `internal` is unit-testable, and should be unit tested.

Reply:
This class is only used by `ConcurrencyUtils`, which iteself is unit-tested, therefore using code coverage tools, we can see that it is tested already. Is that sufficient, or do you want more specific tests?

Reply:
`KeyedMutex` is a standalone, reusable class.  It should be unit tested by itself.